src/actions/loadData.js

```diff
@@ -240,6 +240,47 @@ export const loadSecondTree = (secondTreeUrl, firstTreeUrl) => async (dispatch,
 };
 
 
+const loadMultipleBlocks = (blocks) => async (dispatch) => {
+  const jsons = {};
+  const namesMainTree = [];
+  const namesTreeToo = [];
+
+  for (let i = 0; i < blocks.length; i++) {
+    namesMainTree[i] = blocks[i].dataset.split(":")[0];
+    if (namesMainTree[i]) {
+      // eslint-disable-next-line no-await-in-loop
+      if (!jsons[namesMainTree[i]]) {jsons[namesMainTree[i]] = await (await getDataset(namesMainTree[i])).json();}
+    }
+    namesTreeToo[i] = blocks[i].dataset.split(":")[1];
+    if (namesTreeToo[i]) {
+      // eslint-disable-next-line no-await-in-loop
+      if (!jsons[namesTreeToo[i]]) {jsons[namesTreeToo[i]] = await (await getDataset(namesTreeToo[i])).json();}
+    }
+
+    const state = createStateFromQueryOrJSONs({
+      json: jsons[namesMainTree[i]],
+      secondTreeDataset: jsons[namesTreeToo[i]] || false,
+      mainTreeName: namesMainTree[i],
+      secondTreeName: namesTreeToo[i] || false,
+      narrativeBlocks: blocks,
+      query: { n: i },
+      dispatch
+    });
+
+    if (i === 0) {
+      dispatch({
+        type: types.CLEAN_START,
+        pathnameShouldBe: window.location.pathname,
+        ...state
+      });
+    }
+
+    window[`allBaseStates.${i}`] = state;
```

Comment:
Is there a reason this is better stored as a global rather than a (new piece of) redux state?

I also see potential issues associating it with the block number as in a narrative with blocks 1&2 being dataset A and blocks 3&4 being dataset B, then the user can jump straight from block 1 to block 4 (via the blue dots in the sidebar).

An alternative approach would be to fetch the JSONs behind each of the different datasets in a narrative and store them (as a global, or as redux state). Then, when going between blocks where the dataset differs, we run `createStateFromQueryOrJSONs` and replace the current redux state accordingly. This should be more memory efficient also.

Reply:
See comments below, the state saving system is completely contained within one function, putting it into redux would make it less rather than more maintainable imho